src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -1561,6 +1619,58 @@ bool emberAfThermostatClusterCommitPresetsSchedulesRequestCallback(
     return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::Success);
 }
 
+bool handleAtomicRollback(chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+                          const Commands::AtomicRequest::DecodableType & commandData)
+{
+    if (!validAtomicAttributes(commandData))
+    {
+        commandObj->AddStatus(commandPath, imcode::InvalidCommand);
+        return true;
+    }
+    EndpointId endpoint = commandPath.mEndpointId;
+    bool inAtomicWrite  = gThermostatAttrAccess.InAtomicWrite(commandObj, endpoint);
+    if (!inAtomicWrite)
+    {
+        commandObj->AddStatus(commandPath, imcode::InvalidInState);
+        return true;
+    }
+
+    Delegate * delegate = GetDelegate(endpoint);
+
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        commandObj->AddStatus(commandPath, imcode::InvalidInState);
+        return true;
+    }
+    return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::Success);
```

Comment:
Again, all these SendResponseAndCleanUp calls are returning the wrong sort of response, unless I am really missing something.